src/src/lms/lms.service.ts

```diff
@@ -0,0 +1,435 @@
+				query: query,
+			});
+			if (query_response?.data?.insert_Lms_test_tracking_one) {
+				//create score detail
+				//called without await
+				this.createScoreDetails(
+					lmsTestTrackingDto,
+					query_response,
+					user_id,
+				);
+				return response.status(200).send({
+					success: true,
+					message: 'CreateTestTracking created successfully!',
+					data: query_response.data.insert_Lms_test_tracking_one,
+				});
+			} else {
+				return response.status(404).send({
+					success: false,
+					message: 'Error in CreateTestTracking!',
+					error: query_response?.data,
+				});
+			}
+		} catch (error) {
+			console.log(`CreateTestTracking '. Error!\n`, error.stack);
+			return response.status(404).send({
+				success: false,
+				message: 'Error in CreateTestTracking!',
+				error: error,
+			});
+		}
+	}
+
+	public async createScoreDetails(
+		lmsTestTrackingDto: LSMTestTrackingDto,
+		query_response: any,
+		user_id: any,
+	) {
+		try {
+			let testId = query_response.data.insert_Lms_test_tracking_one.id;
+			let score_detail = lmsTestTrackingDto['score_details'];
+			let scoreObj = [];
+			for (let i = 0; i < score_detail.length; i++) {
+				let section = score_detail[i];
+				let itemData = section?.data;
+				if (itemData) {
+					for (let j = 0; j < itemData.length; j++) {
+						let dataItem = itemData[j];
+						scoreObj.push({
+							user_id: user_id,
+							test_id: testId,
+							question_id: dataItem?.item?.id,
+							pass: dataItem?.pass,
+							section_id: dataItem?.item?.sectionId,
+							max_score: dataItem?.item?.maxscore,
+							score: dataItem?.score,
+							res_value: dataItem?.resvalues
+								? JSON.stringify(dataItem.resvalues)
+								: '',
+							duration: dataItem?.duration,
+						});
+					}
+				}
+			}
+			let data_score_details = {
+				query: `mutation insert_multiple_Lms_score_details($objects: [Lms_score_details_insert_input!]!) {
+					  insert_Lms_score_details(objects: $objects) {
+						returning {
+						  id
+						}
+					  }
+					}
+					`,
+				variables: {
+					objects: scoreObj,
+				},
+			};
+			//insert multiple items
+			let query_score_response =
+				await this.hasuraService.queryWithVariable(data_score_details);
+			if (query_score_response?.data?.data?.insert_Lms_score_details) {
+				//CreateScoreDetail success
+			} else {
+				//Error in CreateScoreDetail!
+				//call again
+				this.createScoreDetails(
+					lmsTestTrackingDto,
+					query_response,
+					user_id,
+				);
+			}
+		} catch (e) {
+			//Error in CreateScoreDetail!
+			//call again
+			this.createScoreDetails(
+				lmsTestTrackingDto,
+				query_response,
+				user_id,
+			);
+		}
+	}
+
+	//get link will be like http://localhost:5000/lms/test/f7185760-bd82-47f2-9b56-6c9777ca0bd4
+	//here f7185760-bd82-47f2-9b56-6c9777ca0bd4 is id from table Lms_test_tracking
+	public async getTestTracking(id: any, header, response) {
+		const userDetail: any = await this.userService.ipUserInfo(header);
```

Comment:
you can get userid from req.mw_user_id , chk MW once @rushi-tekdi and update code accordingly